R/module_teal.R

```diff
@@ -112,7 +112,7 @@ ui_teal <- function(id,
         footer,
         teal.widgets::verbatim_popup_ui(ns("sessionInfo"), "Session Info", type = "link"),
         br(),
-        downloadLink(ns("lockFile"), "Download .lock file"),
+        shinyjs::hidden(downloadLink(ns("lockFile"), "Download .lock file")),
```

Comment:
It goes a little bit beyond this PR but you actually changes this :)

I think it's a bad user experience to be in the state where you don't know if something is running. We should expose that information somewhere somehow. Right now that's the case - if you don't see the link you don't know if it is still doing snapshot or it's a bug or something else.
I would propose to be more transparent in what's happening. That is: start empty. If the process is invoked - change the status to "running" state. If it's completed - present the download link. If it's error - give this to UI. I hope you get what I mean here.

Reply:
You mean, the button that is finally used for the download-handler could have multiple phases/steps: empty, running, ready for download, error?